src/en/mangasaki/src/eu/kanade/tachiyomi/extension/en/mangasaki/MangaSaki.kt

```diff
@@ -0,0 +1,269 @@
+package eu.kanade.tachiyomi.extension.en.mangasaki
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class MangaSaki : ParsedHttpSource() {
+
+    override val name = "MangaSaki"
+
+    override val baseUrl = "https://www.mangasaki.org"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    private var searchMode: Boolean = false
+
+    // popular
+    override fun popularMangaRequest(page: Int): Request {
+        return if (page >= 1) {
+            GET("$baseUrl/directory/hot?page=${page - 1}/", headers)
+        } else {
+            GET("$baseUrl/directory/hot?page=0", headers)
+        }
+    }
```

Comment:
```suggestion
    override fun popularMangaRequest(page: Int): Request {
        return GET("$baseUrl/directory/hot?page=${page - 1}", headers)
    }
```
Pages are guaranteed to start from 1, no need to be defensive.